src/components/Pokedex/PokeCard/PokeCard.tsx

```diff
@@ -1,13 +1,47 @@
-import React from "react";
-import "./PokeCard.css";
-
-const PokeCard = (props: any) => {
-    return (
-        <div className="pokecard">
-            <img src={props.url} alt="Default for Pokemon"/>
-            <p>{props.name}</p>
-        </div>
-    )
+// @ts-nocheck
+import React, { useEffect, useState } from 'react';
+import './PokeCard.css';
+
+function useSinglePokemonDetails(pokemon: any) {
+  const [details, setDetails] = useState(null);
+
+  useEffect(
+    function () {
+      async function fetchDetails() {
+        try {
+          const pokemonDetailsResponse = await fetch(pokemon.url);
+
```

Comment:
I see you using this pattern of setting state and fetching data as close to the component as possible.  Thumbs up here